app/src/main/java/ayds/winchester/songinfo/home/view/ResultReleaseDate.kt

```diff
@@ -0,0 +1,67 @@
+package ayds.winchester.songinfo.home.view
+
+import ayds.winchester.songinfo.home.model.entities.EmptySong
+import ayds.winchester.songinfo.home.model.entities.Song
+
+interface ResultReleaseDate{
+    fun getDescription(song: Song = EmptySong): String
+}
+
+internal class ResultReleaseDateImpl(): ResultReleaseDate{
+
+    override fun getDescription(song: Song): String {
+        return when (song.releaseDatePrecision) {
+            "year" -> getDescriptionByYear(song)
+            "month" -> getDescriptionByMonth(song)
+            "day" -> getDescriptionByDay(song)
+            else -> "Invalid Precision"
+        }
+    }
+
+    private fun getDescriptionByYear(song: Song): String {
+        var fecha = song.releaseDate.split("-").first()
+        return if (!isLeapYear(fecha.toInt())) {
+            "$fecha (not a leap year)"
+        } else "$fecha (leap year)"
+    }
+
+    private fun getDescriptionByMonth(song: Song): String {
+        var fecha = ""
+        if (song.releaseDate.split("-").component2() != "") {
+            fecha = fromNumberToMonth(song.releaseDate.split("-").component2())
+        }
+        return "$fecha, " + song.releaseDate.split("-").first()
+    }
+
+    private fun fromNumberToMonth(month:String):String{
+        var fecha = ""
+        when (month) {
+            "01" ->  fecha = "January"
+            "02" ->  fecha ="February"
+            "03" ->  fecha ="March"
+            "04" ->  fecha ="April"
+            "05" ->  fecha ="May"
+            "06" ->  fecha ="June"
+            "07" ->  fecha ="July"
+            "08" ->  fecha ="August"
+            "09" ->  fecha ="September"
+            "10" ->  fecha ="October"
+            "11" ->  fecha ="November"
+            "12" ->  fecha ="December"
+        }
+        return fecha
```

Comment:
Se puede simplificar:
```suggestion
      return when (month) {
            "01" ->  "January"
            "02" ->  "February"
            "03" ->  "March"
            "04" ->  "April"
            "05" ->  "May"
            "06" ->  "June"
            "07" ->  "July"
            "08" ->  "August"
            "09" ->  "September"
            "10" ->  "October"
            "11" ->  "November"
            "12" ->  "December"
        }
```